detekt-rules-exceptions/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ReturnFromFinally.kt

```diff
@@ -10,12 +10,18 @@ import io.gitlab.arturbosch.detekt.api.Severity
 import org.jetbrains.kotlin.psi.KtFinallySection
 import org.jetbrains.kotlin.psi.KtNamedFunction
 import org.jetbrains.kotlin.psi.KtReturnExpression
+import org.jetbrains.kotlin.psi.KtTryExpression
 import org.jetbrains.kotlin.psi.psiUtil.collectDescendantsOfType
 import org.jetbrains.kotlin.psi.psiUtil.parents
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.bindingContextUtil.isUsedAsExpression
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
 
 /**
  * Reports all `return` statements in `finally` blocks.
  * Using `return` statements in `finally` blocks can discard and hide exceptions that are thrown in the `try` block.
+ * Also report unaffected return value in `finally` block if it used as expression.
+ * See: https://kotlinlang.org/docs/reference/exceptions.html#try-is-an-expression
```

Comment:
This might be better understandable to first time users of this rule.

```suggestion
 * Reports all `return` statements in `finally` blocks.
 * Using `return` statements in `finally` blocks can discard and hide exceptions that are thrown in the `try` block.
 * 
 * Furthermore, this rule reports values from `finally` blocks, if the corresponding `try` is used as an expression.
```